tests/lib/util/glob-util.js

```diff
@@ -221,6 +221,50 @@ describe("globUtil", function() {
             ]);
         });
 
+        it("should silently ignore default ignored files if not passed explicitly", function() {
+            var directory = getFixturePath("glob-util", "hidden");
+            var patterns = [directory];
+            var result = globUtil.listFilesToProcess(patterns, {
+                cwd: getFixturePath()
+            });
+
+            assert.equal(result.length, 0);
+        });
+
+        it("should ignore and warn for default ignored files when passed explicitly", function() {
```

Comment:
I think we may be getting mixed up on a few related issues.  This PR is to fix https://github.com/eslint/eslint/issues/5547, which is to say, to keep ignoring default-ignored files even if the `--no-ignore` flag is passed.  

The current behavior has always been to print a warning if an explicitly passed file is ignored due to an ignore rule.  https://github.com/eslint/eslint/issues/4828#issuecomment-194535273 is a separate issue and will need greater structural changes to accomplish.